### DISCUSSION TITLE
example plugin: how to change parent block in the right way?

### DISCUSSION THREAD

**Discussion by ColonelHathii** [woocommerce/woocommerce#45000] [WooCommerce, Blocks, and Site Editing]
Hi 👋 

We used [extend-cart-checkout-block](https://www.npmjs.com/package/@woocommerce/extend-cart-checkout-block) as a starting point.
In "`src/js/checkout-newsletter-subscription-block/block.json`" we changed: 

`"parent": ["woocommerce/checkout-contact-information-block"],`
to 
`"parent": ["woocommerce/checkout-fields-block"],`

to allow the block to be displayed within the checkout fields and we changed 
`"move": true`
to 
`"move": false`
to allow the block to be moved.

This works fine in the editor:
![screen-1](https://github.com/woocommerce/woocommerce-blocks/assets/143522127/ed1a5abc-0e50-432b-b637-329a65a3b8bd)

But in the frontend the block is always displayed at the bottom of the checkout block:

![screen-2](https://github.com/woocommerce/woocommerce-blocks/assets/143522127/422b14a2-b56f-4990-9340-6604c35484e6)

Why is this, what do we need to pay attention to?

Thanks in advance.

Cheers 🍺

**Comment by opr** [comment]
Hi @ColonelHathii 

You can use this filter:

```php
add_filter(
  '__experimental_woocommerce_blocks_add_data_attributes_to_namespace',
  function() {
    return ['your_block_namespace'];
  },
  10,
  0
);
```

However if you do this you'll need to ensure the inner text for the checkbox label is saved as an attribute. Unless it's saved as an attribute it won't render.

**Reply by ColonelHathii** [reply]
Hi @opr 👋 

thank you very much for your answer. That works!

We would never have thought of that ourselves, that you can achieve the solution with this PHP filter.

It would be helpful to have an example with this filter in [extend-cart-checkout-block](https://www.npmjs.com/package/@woocommerce/extend-cart-checkout-block) or to point out this behavior in a tutorial or documentation. 

Thanks for your time and reply.

Cheers 🍻
